garage/tf/optimizers/penalty_lbfgs_optimizer.py

```diff
@@ -57,6 +57,10 @@ def update_opt(self,
         :param inputs: A list of symbolic variables as inputs
         :return: No return value.
         """
+        constraint_term, constraint_value = leq_constraint
+        penalty_var = tf.placeholder(tf.float32, tuple(), name="penalty")
+        penalized_loss = loss + penalty_var * constraint_term
```

Comment:
I see these variables are obtained again in the lines of code below, but under a scope. Why are they required outside the scope?

Reply:
Oh my bad. I didn't see they are obtained under a scope.

Reply:
But isn't `self._max_constraint_val = constraint_value` requires the line 60?

Reply:
Oh you're right, and using git blame it seems I'm responsible for that. Could you please change the code to:
```
        params = target.get_params(trainable=True)
        with tf.name_scope(name, "PenaltyLbfgsOptimizer",
                           [leq_constraint, loss, params]):
            constraint_term, constraint_value = leq_constraint
            penalty_var = tf.placeholder(tf.float32, tuple(), name="penalty")
            penalized_loss = loss + penalty_var * constraint_term

            self._target = target
            self._max_constraint_val = constraint_value
            self._constraint_name = constraint_name

            def get_opt_output():
```
Sorry for the inconvenience.